models/clm/model_mod.f90

```diff
@@ -4644,13 +3107,547 @@ subroutine say(what)
+endif
+
+end subroutine parse_dimensions
+
+
+!-------------------------------------------------------------------------------
+!>
+
+subroutine loop_other_fastest(varname, indx, dimension_names, dimension_lengths, &
+                                      horiz_dimension, other_dimension)
+character(len=*), intent(in) :: varname
+integer(i8),          intent(inout) :: indx
+character(len=*), intent(in) :: dimension_names(:)
+integer,          intent(in) :: dimension_lengths(:)
+integer,          intent(in) :: horiz_dimension
+integer,          intent(in) :: other_dimension
+
+integer :: i,j,xi,xj
+
+COLUMN : do j = 1, dimension_lengths(horiz_dimension)
+
+   call fill_levels(varname, j, dimension_names(  other_dimension), & 
+                                dimension_lengths(other_dimension),levtot)
+
+   xi = cols1d_ixy(j)
+   xj = cols1d_jxy(j) ! nnnnn_jxy(:) always 1 if unstructured
+   OTHER :  do i = 1, dimension_lengths(other_dimension)
+      levels(  indx) = levtot(i)
+      if (unstructured) then
+         lonixy(  indx) = xi
+         latjxy(  indx) = xi
+         landarea(indx) = AREA1D(xi) * LANDFRAC1D(xi) * cols1d_wtxy(j)
+      else
+         lonixy(  indx) = xi
+         latjxy(  indx) = xj
+         landarea(indx) = AREA2D(xi,xj) * LANDFRAC2D(xi,xj) * cols1d_wtxy(j)
+      endif
+      indx = indx + 1
+   enddo OTHER
+enddo COLUMN
+
+end subroutine loop_other_fastest
+
+
+!-------------------------------------------------------------------------------
+!>
+
+subroutine loop_column_fastest(varname, indx, dimension_names, dimension_lengths, &
+                                      horiz_dimension, other_dimension)
+character(len=*), intent(in) :: varname
+integer(i8),          intent(inout) :: indx
+character(len=*), intent(in) :: dimension_names(:)
+integer,          intent(in) :: dimension_lengths(:)
+integer,          intent(in) :: horiz_dimension
+integer,          intent(in) :: other_dimension
+
+integer :: i,j,xi,xj
+
+OTHER : do j = 1, dimension_lengths(other_dimension)
+   COLUMN :  do i = 1, dimension_lengths(horiz_dimension)
+
+      xi = cols1d_ixy(i)
+      xj = cols1d_jxy(i) ! nnnnn_jxy(:) always 1 if unstructured
+
+      call fill_levels(varname, i, dimension_names(  other_dimension), & 
+                                   dimension_lengths(other_dimension),levtot)
+
+      levels(  indx) = levtot(j)
+      if (unstructured) then
+         lonixy(  indx) = xi
+         latjxy(  indx) = xi
+         landarea(indx) = AREA1D(xi) * LANDFRAC1D(xi) * cols1d_wtxy(i)
+      else
+         lonixy(  indx) = xi
+         latjxy(  indx) = xj
+         landarea(indx) = AREA2D(xi,xj) * LANDFRAC2D(xi,xj) * cols1d_wtxy(i)
+      endif
+      indx = indx + 1
+   enddo COLUMN
+enddo OTHER
+
+end subroutine loop_column_fastest
+
+
 !===================================================================
 ! End of model_mod
 !===================================================================
+
 end module model_mod
```

Comment:
ABSOLUTELY ... the file _changes_ should no be reviewed. When I get the scripting finished, it will be possible to test the model_mod in a more rigorous fashion. Without a diverse set of input files, it is hard to test "everything".